### PR TITLE
Add date range to ingestion load reports

### LINKED ISSUE #468: Include date information in loading completion message
## Problem

Our load completion reporting Slack message currently does not include any date information. For instances where `catchup=True` and we need to re-run multiple days, it can be hard to distinguish which runs came from which days.

## Description

It would be nice to have a date range specified for each DAG. E.g. `@daily` would have just the day it ran on, say `2022-04-08`. `@weekly` could be `2022-04-01 -> 2022-04-08`. DAGs that consume the entire API could say `date: all` or something.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by AetherUnbound**
## Fixes

Fixes #468 by @AetherUnbound

## Description

This PR adds an additional section to the reported ingestion metrics slack message which also includes the date range for the data ingested. This date range only shows up for `dated` DAGs, as other DAGs are assumed to ingest the entire dataset on each scheduled run.

Screenshot_2022-07-19_13-24-34

## Testing Instructions

Run a DAG and see that the completion message includes dated information!

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: openverse_catalog/dags/providers/provider_dag_factory.py [hunk 1]
```diff
@@ -316,6 +316,9 @@ def create_provider_api_workflow(
                 "provider_name": provider_name,
                 "duration": XCOM_PULL_TEMPLATE.format(pull_data.task_id, "duration"),
                 "record_counts_by_media_type": record_counts_by_media_type,
+                "dated": dated,
+                "date_range_start": "{{ data_interval_start | ds }}",
+                "date_range_end": "{{ data_interval_end | ds }}",
```
Comment (stacimc) [inline openverse_catalog/dags/providers/provider_dag_factory.py:321]: TIL, so nice that it's so easy to get the interval start/end!

File: tests/dags/common/loader/test_reporting.py [hunk 2]
```diff
@@ -95,15 +95,45 @@ def _make_report_completion_contents_data(media_type: str):
 @pytest.mark.parametrize(
     "image_data, image_expected", _make_report_completion_contents_data("image")
 )
+@pytest.mark.parametrize(
+    "dated, date_range_start, date_range_end, expected_date_range",
+    [
+        # Not dated, no date range
+        (False, "2022-01-01", "2022-05-01", "all"),
+        # Not dated, but date range supplied
+        (False, None, None, "all"),
```
Comment (stacimc) [inline tests/dags/common/loader/test_reporting.py:104]: Not a big deal, but I think the two comments above are switched around.

**Review by stacimc — APPROVED** [review]
Love it! 🥳 I tested with Wikimedia Commons and with Cleveland Museum to verify dated and non-dated. Looks great :)

**Review by sarayourfriend — APPROVED** [review]
Cool!

File: tests/dags/common/loader/test_reporting.py (hunk 2 above)
Comment (AetherUnbound) [reply tests/dags/common/loader/test_reporting.py:104]: Whoops! How'd that happen 😂 thanks!

File: openverse_catalog/dags/providers/provider_dag_factory.py (hunk 1 above)
Comment (AetherUnbound) [reply openverse_catalog/dags/providers/provider_dag_factory.py:321]: Airflow's template & macros are _so_ powerful!

**Pull request merged**